### ISSUE TITLE
Two endpoints are handling the request when use DynamicRouteValueTransformer

### ISSUE THREAD

**Issue by sefacan** [dotnet/aspnetcore#28702]
### Describe the bug
Two endpoints are handling the request when use DynamicRouteValueTransformer
When I try to go "/contact-us" page, "ContactUs" action is invoking and PageRouteTransformer is handling the request.

By the way, All trasformers are handling static files. I'm using the ContentTypeFileProvider for skip handling but it has a big freaking dictionary for mime-types. It's not suitable for performance.

### To Reproduce
```
// contact us
endpoints.MapControllerRoute("ContactUs",
    "contact-us",
    new { controller = "Page", action = "ContactUs" });

// pages
endpoints.MapControllerRoute("Page",
    "{sename}",
    new { controller = "Page", action = "Detail" });

// for dynamic pages
endpoints.MapDynamicControllerRoute<PageRouteTransformer>("{sename}");
```

### Further technical details
.NET SDK (reflecting any global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.101\

Host (useful for support):
  Version: 5.0.1
  Commit:  b02e13abab

.NET SDKs installed:
  2.2.207 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.402 [C:\Program Files\dotnet\sdk]
  3.1.404 [C:\Program Files\dotnet\sdk]
  5.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
 

Microsoft Visual Studio Community 2019 Version 16.8.3

**Comment by pranavkm** [comment]
FYI @javiercn

**Comment by omfgicbf** [comment]
Is this the same as #21083?

In #21083 it seems to have been an issue in 3.1 that was addressed as part of #25073, so it's not clear if:
- It was fixed and should work now but doesn't,
- It's just not expected to work yet and is being considered for a future release, or
- This is a different issue
